shared/actions/chat/inbox.js

```diff
@@ -2,141 +2,279 @@
-  const finalizedState: Constants.FinalizedState = _inboxToFinalized(inbox)
+  const conversations: List<Constants.InboxState> = List((inbox.conversationsUnverified || []).map(c => {
+    if (c.metadata.visibility !== ChatTypes.CommonTLFVisibility.private) { // private chats only
+      return null
+    }
+
+    const msgMax = c.maxMsgSummaries && c.maxMsgSummaries.length && c.maxMsgSummaries[0]
+    if (!msgMax || msgMax.tlfName.includes('#')) { // We don't support mixed reader/writers
+      return null
+    }
+
+    return new Constants.InboxStateRecord({
+      conversationIDKey: Constants.conversationIDToKey(c.metadata.conversationID),
+      info: null,
+      participants: List(parseFolderNameToUsers(author, msgMax.tlfName).map(ul => ul.username)),
+      snippet: ' ',
+      state: 'untrusted',
+      status: Constants.ConversationStatusByEnum[c.metadata.status || 0],
+      time: c.readerInfo && c.readerInfo.mtime,
+    })
+  }).filter(Boolean))
 
   yield put(Creators.loadedInbox(conversations))
-  if (finalizedState.count()) {
-    yield put(Creators.updateFinalizedState(finalizedState))
+  chatInboxUnverified.response.result()
+}
+
+// Unbox one specific item by keys
+function * getInboxAndUnbox ({payload: {conversationIDKeys}}: Constants.GetInboxAndUnbox): SagaGenerator<any, any> {
+  const param: ChatTypes.localGetInboxAndUnboxLocalRpcParam = {
+    identifyBehavior: TlfKeysTLFIdentifyBehavior.chatGui,
+    query: {..._getInboxQuery, readOnly: true},
   }
 
-  chatInboxUnverified.response.result()
+  const result: ChatTypes.GetInboxAndUnboxLocalRes = yield call(ChatTypes.localGetInboxAndUnboxLocalRpcPromise, {param})
+  if (result.conversations) {
+    yield result.conversations.map(conv => call(updateInbox, conv))
+  }
+}
 
-  let finishedCalled = false
-  while (!finishedCalled) {
-    const incoming: {[key: string]: any} = yield race({
-      chatInboxConversation: Saga.takeFromChannelMap(loadInboxChanMap, 'chat.1.chatUi.chatInboxConversation'),
-      chatInboxFailed: Saga.takeFromChannelMap(loadInboxChanMap, 'chat.1.chatUi.chatInboxFailed'),
-      finished: Saga.takeFromChannelMap(loadInboxChanMap, 'finished'),
-      timeout: call(delay, 30000),
-    })
+function _toSupersedeInfo (conversationIDKey: Constants.ConversationIDKey, supersedeData: Array<ChatTypes.ConversationMetadata>): ?Constants.SupersedeInfo {
+  const parsed = supersedeData
+    .filter(md => md.idTriple.topicType === ChatTypes.CommonTopicType.chat && md.finalizeInfo)
+    .map(md => ({
+      conversationIDKey: Constants.conversationIDToKey(md.conversationID),
+      finalizeInfo: unsafeUnwrap(md && md.finalizeInfo),
+    }))
+  return parsed.length ? parsed[0] : null
+}
 
-    if (incoming.chatInboxConversation) {
-      requestIdleCallback(() => {
-        incoming.chatInboxConversation.response.result()
-      }, {timeout: 100})
+// Update an unboxed inbox item
```

Comment:
It's actually not unboxed because you can get here on rekey from below? https://github.com/keybase/client/pull/6324/files#diff-4efbf83cd5610da824ade03bfb4d3737R277